sematic/resolvers/worker.py

```diff
@@ -135,7 +135,7 @@ def main(run_id: str, resolve: bool):
             future: Future = func(**kwargs)
             future.id = run.id
 
-            resolver = CloudResolver(detach=False)
+            resolver = CloudResolver(detach=False, is_running_remotely=True)
             resolver.set_graph(runs=runs, artifacts=artifacts, edges=edges)
 
             resolver.resolve(future)
```

Comment:
I know this is already the second iteration of this mechanism so I'm happy to go with that, but just for the sake of #seektruth, is there a more elegant way where
- we don't need `is_running_remotely`
- and instead of calling `resolver.resolve` on #141, we call a new method, e.g. `resolve_remotely` or `resolve_on_worker`, which would not call `_resolution_will_start` and therefore the resolution would not get created...

This would remove the need for `is_running_remotely` and for overriding `_create_resolution` on `CloudResolver`.

The fact that I used `.resolve` here is purely a shortcut, but it would make sense to call something else that does not do exactly the same as an actual call to `.resolve`, which usually means a user triggered a resolution.

Reply:
Not without duplicating some code or passing something akin to `is_running_remotely` in somewhere. Imagine what the method setup would be in this proposed setup:

## Option A
```
def resolve(self):
    self._do_real_resolve(is_running_remotely=False)
    
def resolve_remotely(self):
    self._do_real_resolve(is_running_remotely=True)
```

## Option B
```
def resolve(self):
    # resolve logic, and call resolution_will_start
    
def resolve_remotely(self):
    # resolve logic, but don't call resolution_will_start
```

I don't really like either of these options more than what is done here. B is clearly worse because it duplicates a bunch of logic. A is better than B, but not necessarily better than what we have. It has a few limitations that make me think it might be slightly worse: you are passing an argument called `is_running_remotely` from `.resolve()`, which means it is also getting passed to the Silent/Local resolvers. These would never run remotely, so it doesn't really make sense.

You might be thinking "what about Option C like this?"

```
# in state_machine_resolver
def resolve():
    # normal resolution logic
    
# in cloud_resolver
def resolve():
    super().resolve()
    # some customization?
```

But this doesn't really help/work. You either need to pass some argument into `resolve` to change its behavior, or you need to replicate a bunch of logic, minus the "resolution will start"

Reply:
Ok, makes sense, thanks for taking the time to think through this.